packages/editor/src/core/Provider/index.tsx

```diff
@@ -161,6 +163,7 @@ const Provider: React.FC<ProviderProps> = ({
     JSON.stringify(childConstraints ?? {}), // its an object, we prevent unnecessary rerenders by stringify it
     ...Object.keys(components ?? {}),
     ...Object.values(components ?? {}), // minimize unnecessary rerenders by forcing shallow comparison of "components" object
+    cellSpacing,
```

Comment:
if its a number, this works, but it might lead to rerenders if it is an array or object (if we e.g. want to allow to have dfifferent vertical and horizontal padding)

JSON.stringify might be better here

Reply:
I was going to do something like `Array.isArray(cellSpacing) ? ...cellSpacing : cellSpacing` but it looks like TS doesn't recognize `Array.isArray` as a type guard. JSON.stringify is going to be shorter.